api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1650,4 +1650,13 @@ private void withCurrentUserAs(User user, Runnable internals) throws IllegalAcce
 			FieldUtils.getField(UserContext.class, "user", true).set(userContext, authenticatedUser);
 		}
 	}
+
+	@Test
+	public void saveUserProperty_shouldAddANewPropertyWithAVeryLargeStringWithoutRunningIntoError() {
+		final String USER_PROPERTY_KEY = liquibase.util.StringUtil.repeat("emrapi.lastViewedPatientIds,",10);
+		final String USER_PROPERTY_VALUE = liquibase.util.StringUtil.repeat("52345",9899);
+		User updatedUser = userService.saveUserProperty(USER_PROPERTY_KEY, USER_PROPERTY_VALUE);
+		assertEquals(280, updatedUser.getUserProperties().keySet().stream().findFirst().get().length());
```

Comment:
@dkayiwa this solves the previous useless looping.

Reply:
How many items does this collection have?

Reply:
The collection has only one item.

Reply:
How do you compare the above with? `keySet().iterator().next()`?

Reply:
True , i think it's a better option .